### PR TITLE
Implements PartialOrd for Number.

### LINKED ISSUE #328: Impl PartialOrd for Number
It seems possible to implement both of these for Number.

### PR DESCRIPTION
**Pull request by remexre**
Resolves #328.

### CODE REVIEW

File: src/number.rs [hunk 1]
```diff
@@ -237,6 +238,40 @@ impl Debug for Number {
     }
 }
 
+impl PartialOrd for Number {
+    fn partial_cmp(&self, other: &Number) -> Option<Ordering> {
+        fn reverse(o: Ordering) -> Ordering {
```
Comment (clarcharr) [inline src/number.rs]: Why not use `Ordering::reverse` instead?

Comment (remexre) [inline src/number.rs]: Oh, didn't know that was a thing.

**Comment by remexre** [comment]
Okay, fixed the reverse

**Comment by clarcharr** [comment]
You also probably should add tests for this. Also, I have a feeling that we'd still be able to turn this into an `Ord` definition with a bit of work; I'll try and see what I can do.

**Comment by remexre** [comment]
`Ord` required `Eq`, though, and `f64`s are only `PartialEq` (because `NaN`s). Since we check that `NaN` never appears, it is legal to make this type `Eq` and `Ord`. I'll add the tests if this works (unrelated tests seem to be failing locally; it might just be the nightly I'm on, so I'm gonna let Travis check it).

**Pull request closed without merging**